src/AD_ForwardDiff.jl

```diff
@@ -18,28 +18,30 @@ end
 
 _default_chunk(ℓ) = ForwardDiff.Chunk(dimension(ℓ))
 
-function _default_gradientconfig(ℓ, chunk::ForwardDiff.Chunk)
-    ForwardDiff.GradientConfig(Base.Fix1(logdensity, ℓ), zeros(dimension(ℓ)), chunk)
+function _default_gradientconfig(ℓ, chunk::ForwardDiff.Chunk, x::AbstractVector)
+    return ForwardDiff.GradientConfig(Base.Fix1(logdensity, ℓ), x, chunk)
 end
 
-function _default_gradientconfig(ℓ, chunk::Integer)
-    _default_gradientconfig(ℓ, ForwardDiff.Chunk(chunk))
+function _default_gradientconfig(ℓ, chunk::Integer, x::AbstractVector)
+    return _default_gradientconfig(ℓ, ForwardDiff.Chunk(chunk), x)
 end
 
 """
-    ADgradient(:ForwardDiff, ℓ; chunk, gradientconfig)
-    ADgradient(Val(:ForwardDiff), ℓ; chunk, gradientconfig)
+    ADgradient(:ForwardDiff, ℓ; input, chunk, gradientconfig)
+    ADgradient(Val(:ForwardDiff), ℓ; input, chunk, gradientconfig)
 
 Wrap a log density that supports evaluation of `Value` to handle `ValueGradient`, using
 `ForwardDiff`.
 
 Keywords are passed on to `ForwardDiff.GradientConfig` to customize the setup. In
 particular, chunk size can be set with a `chunk` keyword argument (accepting an integer or a
-`ForwardDiff.Chunk`).
+`ForwardDiff.Chunk`), and the underlying vector used by `ForwardDiff` can be set with the
+`input` keyword argument (accepting an `AbstractVector`).
 """
 function ADgradient(::Val{:ForwardDiff}, ℓ;
+                    input = zeros(dimension(ℓ)),
```

Comment:
IMO it is nice to use default values of `nothing` since that clearly indicates if a user provided a custom value and makes it easy to dispatch later and define the default values in the downstream methods were they are actually needed (and possibly changed at some point).
So maybe similar to the ReverseDiff implementation use
```suggestion
                    input::Union{Nothing,AbstractVector} = nothing,
```

Reply:
Hmm, though I see what you mean, it does add two additional methods (we need one for `chunk::Integer` and one for `chunk::FowardDiff.Chunk`).

Reply:
And atm there's no need to know whether the user specified it or not, right?

Reply:
Okay I think I found a "clean" way I'm happy with